lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -105,6 +105,74 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
   helpers are: `live_dashboard_path/2`, `live_dashboard_path/3`,
   `encode_app/1`, `encode_ets/1`, `encode_pid/1`, `encode_port/1`,
   and `encode_socket/1`.
+
+  ## Custom Hooks
+
+  If your page needs to register custom hooks, you can use the `register_after_opening_head_tag/2`
+  function. Because the hooks need to be available on the dead render in the layout, before the
+  LiveView's LiveSocket is configured, your need to do this inside an `on_mount` hook:
+
+  ```elixir
+  defmodule MyAppWeb.MyLiveDashboardHooks do
+    import Phoenix.LiveView
+    import Phoenix.Component
+
+    alias Phoenix.LiveDashboard.PageBuilder
+
+    def on_mount(:default, _params, _session, socket) do
+      {:cont, PageBuilder.register_after_opening_head_tag(socket, &after_opening_head_tag/1)}
+    end
+
+    defp after_opening_head_tag(assigns) do
+      ~H\"\"\"
+      <script>
```

Comment:
What if instead of asking them to modify `window.customHooks` we tell them to add meta tags or attributes to their script?

```suggestion
      <script data-phx-dashboard-hook="MyHook">
```

And then we read all of these hooks somehow? I guess we would need to read from window anyway? 🤔

Reply:
I updated the code to instead provide and use a `window.LiveDashboard.registerCustomHooks({ MyHook: {} })`.

I also clarified the use if CSP nonces.